### PR TITLE
build: separate demo-app in its own directory.

### PR DESCRIPTION
**Pull request by hansl**

### CODE REVIEW

File: src/components/card/card.ts [hunk 1]
```diff
@@ -21,9 +21,10 @@ it also provides a number of preset styles for common card sections, including:
 */
 
 @Component({
+  moduleId: module.id,
   selector: 'md-card',
-  templateUrl: './components/card/card.html',
-  styleUrls: ['./components/card/card.css'],
+  templateUrl: 'ard.html',
```
Comment (robwormald) [inline src/components/card/card.ts]: typo

File: angular-cli-build.js [hunk 2]
```diff
@@ -5,16 +5,53 @@ const path = require('path');
...
 const Funnel = require('broccoli-funnel');
 const MergeTree = require('broccoli-merge-trees');
 const autoPrefixerTree = require('broccoli-autoprefixer');
 
 
 module.exports = function(defaults) {
-  var angularAppTree = new Angular2App(defaults, {
-    sourceDir: 'src/',
+  const appTree = buildAppTree(defaults);
+  const cssAutoprefixed = autoPrefixerTree(new Funnel(appTree, {
```
Comment (jelbourn) [inline angular-cli-build.js]: The changes in this file would use some high-level comments explaining what's going on.

Comment (hansl) [inline angular-cli-build.js]: Added some.

**Comment by jelbourn** [comment]
LGTM

**Comment by jelbourn** [comment]
@hansl looks like the inline-resources task isn't working. I think you just have to update the gulpfile.

**Comment by hansl** [comment]
Don't tell me it doens't support relative path of resources :(

File: scripts/release/inline-resources.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env node
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const glob = require('glob');
+
+/**
+ * Simple Promiseify function that takes a Node API and return a version that supports promises.
+ */
```
Comment (jelbourn) [inline scripts/release/inline-resources.js]: Add comment for why you need this function in this script.

File: scripts/release/inline-resources.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    .filter(name => name.match(/\.js$/));  // Matches only JavaScript files.
+
+  // Generate all files content with inlined templates.
+  files.forEach(filePath => {
+    readFile(filePath, 'utf-8')
+      .then(content => inlineTemplate(filePath, content))
+      .then(content => inlineStyle(filePath, content))
+      .then(content => writeFile(filePath, content))
+      // .then(content => console.log(content))
+      // .then(() => process.exit(0))
```
Comment (jelbourn) [inline scripts/release/inline-resources.js]: Commented lines

File: scripts/release/inline-resources.js [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+ * For every argument, inline the templates and styles under it and write the new file.
+ */
+for (let arg of process.argv.slice(2)) {
+  if (arg.indexOf('*') < 0) {
+    // Argument is a directory target, add glob patterns to include every files.
+    arg = path.join(arg, '**', '*');
+  }
+
+  const files = glob.sync(arg, {})
+    .filter(name => name.match(/\.js$/));  // Matches only JavaScript files.
```
Comment (jelbourn) [inline scripts/release/inline-resources.js]: Should use either `RegExp.prorotype.test` or `String.prototype.search` rather than `match`

**Comment by jelbourn** [comment]
You're also got a lint error.

File: scripts/release/inline-resources.js (hunk 3 above)
Comment (hansl) [inline scripts/release/inline-resources.js]: Done.

File: scripts/release/inline-resources.js (hunk 4 above)
Comment (hansl) [inline scripts/release/inline-resources.js]: Done.

File: scripts/release/inline-resources.js (hunk 5 above)
Comment (hansl) [inline scripts/release/inline-resources.js]: Done.

**Pull request merged**